src/extensions/step_handlers/script_handler/src/script_handler.cpp

```diff
@@ -572,56 +606,92 @@ static ADUC_Result ScriptHandler_PerformAction(const std::string& action, const
         Log_Debug("##########\n# ADU-SHELL ARGS:\n##########\n %s", ss.str().c_str());
     }
 
-    exitCode = ADUC_LaunchChildProcess(ADUSHELL_FILE_PATH, aduShellArgs, scriptOutput);
+    if (prepareArgsOnly)
+    {
+        std::stringstream ss;
+        for (const auto& a : aduShellArgs)
+        {
+            char quote = '\"';
+            char whitespace = ' ';
+
+            // NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers,readability-magic-numbers)
+            if (a[0] != '-')
+            {
+                // If the argument contains a quote character, then use the other quote character.
```

Comment:
what if there's a mix of single and double quotes, or escaped quotes?

Reply:
yeah, this still doesn't seem robust.